### PR TITLE
Dashboard: Infinite Scroll Component 

### PR DESCRIPTION
**Pull request by BrittanyIRL**
# Overview
Adds an `InfiniteScroller` component for use in dashboard to grab new pages of data while scrolling. 

So in practice InfiniteScroller just gets wrapped around the component we want to be our container for scrolling, then the element to watch enter the screen gets added below the 'children' in the InfiniteScroller. 

## Testing 
- See storybook Dashboard/Components/InfiniteScroller and scroll.
  - If you scroll real fast you can see the 'loading' text 
  - After you reach 30 demo items you see that the component knows that there's no more data to go get and the intersection observer is unobserving. 

## Package additions
`intersection-observer` polyfill - https://github.com/w3c/IntersectionObserver/tree/master/polyfill to make sure all browsers can perform intersection observer for scrolling .

## To Do Separate
- [ ] We need to style the loading and 'no more data' components 
- [ ] Need default language from Sam for these (She's going to do UX for the whole thing later) 
- [ ] "Scroll to Top" functionality (button, visible on page whenever not already at top of page that takes user back to the top)

### CODE REVIEW

**Review by carloskelly13 — CHANGES_REQUESTED** [review]
The code looks great to me. I pulled and tested, works perfectly in Safari and Chrome, but I cannot get it to work at all in Firefox. No console errors either.

File: assets/src/dashboard/components/infiniteScroller/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+} from 'react';
+import PropTypes from 'prop-types';
+import 'intersection-observer';
+
+const InfiniteScroller = ({
+  allDataLoadedMessage = 'No More Stories',
+  children,
+  handleGetData,
+  isAllDataLoaded,
+  loadingMessage = 'Loading...',
```
Comment (carloskelly13) [inline assets/src/dashboard/components/infiniteScroller/index.js]: Let's make sure these use the wp `___` helpers for text

**Review by carloskelly13 — CHANGES_REQUESTED** [review]

File: assets/src/dashboard/components/infiniteScroller/index.js (hunk 1 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/components/infiniteScroller/index.js]: oh good catch!

**Comment by BrittanyIRL** [comment]
> The code looks great to me. I pulled and tested, works perfectly in Safari and Chrome, but I cannot get it to work at all in Firefox. No console errors either.

SO, it turns out that firefox + iframe + intersection observer cannot deduce a default `root` if it's null in the set up for an observer. Since storybook uses iframes, this bug cropped up. The more you know, right? 

Added a ref for the container and am using that as the root and it seems fixed. Will you take a second look?

**Review by carloskelly13 — APPROVED** [review]

**Comment by carloskelly13** [comment]
Looks great in Firefox!

**Review by mariano-formidable — APPROVED** [review]
Looks good to me!  Excellent!! 🎉

**Pull request merged**